x11rb-protocol/src/x11_utils.rs

```diff
@@ -607,6 +607,34 @@ macro_rules! bitmask_binop {
                 *self = *self | Self::from(other)
             }
         }
+        impl core::ops::BitAnd for $t {
+            type Output = $t;
+            fn bitand(self, other: Self) -> Self::Output {
+                Self::from(<$u>::from(self) & <$u>::from(other))
+            }
+        }
+        impl core::ops::BitAnd<$u> for $t {
+            type Output = $t;
+            fn bitand(self, other: $u) -> Self::Output {
+                self & Self::from(other)
+            }
+        }
+        impl core::ops::BitAnd<$t> for $u {
+            type Output = $t;
+            fn bitand(self, other: $t) -> Self::Output {
+                <$t>::from(self) & other
+            }
+        }
+        impl core::ops::BitAndAssign<$t> for $u {
+            fn bitand_assign(&mut self, other: $t) {
+                *self &= Self::from(other)
+            }
+        }
+        impl core::ops::BitAndAssign<$u> for $t {
+            fn bitand_assign(&mut self, other: $u) {
+                *self = *self & Self::from(other)
```

Comment:
Nitpick: Wouldn't it be better to use `&=` here, semantically?

Reply:
I just C-V above `BitOr` codes should I edit both?

Reply:
`*self &= ` Can't be used because that operator'll call `BitAndAssign` which cause infinite loop.

Reply:
I meant over `self.0` or whatever this type is a newtype around.